pytest_fixtures/core/upgrade.py

```diff
@@ -12,3 +15,30 @@ def dependent_scenario_name(request):
         if 'depend_on' in mark.kwargs
     ][0]
     yield depend_test_name
+
+
+@pytest.fixture(scope="session")
+def pre_configured_capsule(session_target_sat):
+    """
+    This fixture returns the Capsule object matching the pre-configured capsule
+
+    This fixture assumes that the Capsule is already integrated with Satellite and
+    the Capsule host is present in the broker inventory.
+    """
+    inventory = {h.hostname for h in Broker().from_inventory()}
+    capsules = {h.name for h in session_target_sat.api.Capsule().search()}
+    # do an intersection of both sets to just select those hosts that we have in the capsules
+    # from that intersection take the Capsule that is not Satellite and return it
+    intersect = inventory.intersection(capsules)
+    intersect.remove(session_target_sat.hostname)
```

Comment:
```suggestion
    intersect.discard(session_target_sat.hostname)
```
remove() raises a KeyError

Reply:
Good point! Will apply the suggestion.